app/src/main/java/org/mozilla/fenix/onboarding/view/UpgradeOnboardingScreen.kt

```diff
@@ -0,0 +1,215 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.view
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.insets.navigationBarsPadding
+import com.google.accompanist.insets.statusBarsPadding
+import mozilla.telemetry.glean.private.NoExtras
+import org.mozilla.fenix.GleanMetrics.Onboarding
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Enum that represents the onboarding page that is displayed.
+ */
+private enum class UpgradeOnboardingState {
+    Welcome,
+    SyncSignIn,
+}
+
+/**
+ * A screen for displaying a welcome and sync sign in onboarding.
+ *
+ * @param isUserSignedIn Whether or not the user is signed into their Firefox Sync account.
+ * @param onDismiss Invoked when the user clicks on the close or "Skip" button.
+ * @param onSignInButtonClick Invoked when the user clicks on the "Sign In" button
+ */
+@Composable
+fun UpgradeOnboardingScreen(
+    isUserSignedIn: Boolean,
+    onDismiss: () -> Unit,
+    onSignInButtonClick: () -> Unit,
+) {
```

Comment:
As gl mentioned in [this comment](https://github.com/mozilla-mobile/fenix/pull/28376/files#r1060997395), the pattern I usually see is to host top-level state and events in the parent fragment. This will make it easier to move these metrics calls into a controller as well. An example of that could be the [WallpaperSettingsFragment](https://searchfox.org/mozilla-mobile/rev/825ba5537cdc9901b5a62bd4e8f9016da29e7c9d/fenix/app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt#61).

Basically, I think we could remove the `UpgradeOnboardingScreen` entirely here, and call directly to `OnboardingContent` though it would likely make sense to change the name as well.